applications/asset_tracker_v2/src/modules/util_module.c

```diff
@@ -199,127 +203,133 @@ static void send_reboot_request(void)
+		send_reboot_request();
+		return;
 	}
 
-	if (is_sensor_module_event(&msg->module.sensor.header)) {
-		switch (msg->module.sensor.type) {
-		case SENSOR_EVT_ERROR:
-			send_reboot_request();
-			break;
-		case SENSOR_EVT_SHUTDOWN_READY:
-			reboot_ack_cnt++;
-			break;
-		default:
-			break;
-		}
+	if (IS_EVENT(msg, modem, MODEM_EVT_ERROR)) {
+		send_reboot_request();
+		return;
 	}
 
-	if (is_gps_module_event(&msg->module.gps.header)) {
-		switch (msg->module.gps.type) {
-		case GPS_EVT_ERROR_CODE:
-			send_reboot_request();
-			break;
-		case GPS_EVT_SHUTDOWN_READY:
-			reboot_ack_cnt++;
-			break;
-		default:
-			break;
-		}
+	if (IS_EVENT(msg, sensor, SENSOR_EVT_ERROR)) {
+		send_reboot_request();
+		return;
 	}
 
-	if (is_data_module_event(&msg->module.data.header)) {
-		switch (msg->module.data.type) {
-		case DATA_EVT_ERROR:
-			send_reboot_request();
-			break;
-		case DATA_EVT_SHUTDOWN_READY:
-			reboot_ack_cnt++;
-			break;
-		default:
-			break;
-		}
+	if (IS_EVENT(msg, gps, GPS_EVT_ERROR_CODE)) {
+		send_reboot_request();
+		return;
 	}
 
-	if (is_app_module_event(&msg->module.app.header)) {
-		switch (msg->module.app.type) {
-		case APP_EVT_ERROR:
-			send_reboot_request();
-			break;
-		case APP_EVT_SHUTDOWN_READY:
-			reboot_ack_cnt++;
-			break;
-		default:
-			break;
-		}
+	if (IS_EVENT(msg, data, DATA_EVT_ERROR)) {
+		send_reboot_request();
+		return;
 	}
 
-	if (is_ui_module_event(&msg->module.ui.header)) {
-		switch (msg->module.ui.type) {
-		case UI_EVT_ERROR:
-			send_reboot_request();
-			break;
-		case UI_EVT_SHUTDOWN_READY:
-			reboot_ack_cnt++;
-			break;
-		default:
-			break;
-		}
+	if (IS_EVENT(msg, app, APP_EVT_ERROR)) {
+		send_reboot_request();
+		return;
 	}
 
-	/* Reboot if after a shorter timeout if all modules has acknowledged
-	 * that the application is ready to shutdown. This ensures a graceful
-	 * shutdown.
-	 */
-	if (reboot_ack_cnt >= module_active_count_get() - 1) {
-		LOG_WRN("All modules have ACKed the reboot request.");
-		LOG_WRN("Reboot in 5 seconds.");
-		k_delayed_work_submit(&reboot_work, K_SECONDS(5));
+	if (IS_EVENT(msg, ui, UI_EVT_ERROR)) {
+		send_reboot_request();
+		return;
 	}
 }
 
-static void message_handler(struct util_msg_data *msg)
+/* Message handler for STATE_REBOOT_PENDING. */
+static void on_state_reboot_pending(struct util_msg_data *msg)
+{
+	if (IS_EVENT(msg, cloud, CLOUD_EVT_SHUTDOWN_READY)) {
+		reboot_ack_check();
+		return;
+	}
+
+	if (IS_EVENT(msg, modem, MODEM_EVT_SHUTDOWN_READY)) {
+		reboot_ack_check();
+		return;
+	}
+
+	if (IS_EVENT(msg, sensor, SENSOR_EVT_SHUTDOWN_READY)) {
+		reboot_ack_check();
+		return;
+	}
+
+	if (IS_EVENT(msg, gps, GPS_EVT_SHUTDOWN_READY)) {
+		reboot_ack_check();
+		return;
+	}
+
+	if (IS_EVENT(msg, data, DATA_EVT_SHUTDOWN_READY)) {
+		reboot_ack_check();
+		return;
+	}
+
+	if (IS_EVENT(msg, app, APP_EVT_SHUTDOWN_READY)) {
+		reboot_ack_check();
+		return;
+	}
+
+	if (IS_EVENT(msg, ui, UI_EVT_SHUTDOWN_READY)) {
```

Comment:
All these ifs can be combined into one, as they all end up calling rebppt_ack_check() and returning

Reply:
I mainly kept it like this in order to more easily expand on functionality specific to single events. I suspect we need to handle the events separately when we introduce the shutdown ACK list. Ill combine the event checks in the ```on_state_init``` function. I don't see any immediate reasons why we should not do it in that case.